sympy/polys/domains/powerseriesring.py

```diff
@@ -0,0 +1,138 @@
+from __future__ import annotations
+
+from sympy.core.expr import Expr
+from sympy.polys.domains.ring import Ring
+from sympy.polys.domains.compositedomain import CompositeDomain
+from sympy.polys.domains.domain import Er, Domain
+from sympy.polys.series.ring import (
+    power_series_ring,
+    PowerSeriesRingRing,
+    PowerSeriesRingField,
+    PowerSeriesElement,
+)
+from sympy.polys.series.tring import TElement
+from sympy.utilities import public
+
+from typing import Union, TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from typing import TypeIs
+
+
+@public
+class PowerSeriesRing(Ring[PowerSeriesElement[Er]], CompositeDomain):
+    """A Domain class for representing univariate power series rings."""
+
+    is_PowerSeriesRing = is_Series = True
+
+    has_assoc_Ring = True
+    has_assoc_Field = False
+
+    ring: Union[PowerSeriesRingRing, PowerSeriesRingField]
```

Comment:
This should be some kind of protocol like:
```
PowerSeriesRingProto[Er]
```
There should be a protocol that both classes match so that the type here is the protocol not a union. The type parameter Er needs to be here.